webapp/channels/src/components/admin_console/session_length_settings.tsx

```diff
@@ -7,33 +7,51 @@ import {FormattedMessage} from 'react-intl';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 import * as Utils from 'utils/utils';
 
-import AdminSettings from './admin_settings';
+import AdminSettings, {BaseState, BaseProps} from './admin_settings';
 import BooleanSetting from './boolean_setting';
-import SettingsGroup from './settings_group.jsx';
+import SettingsGroup from './settings_group.js';
 import TextSetting from './text_setting';
+import { DeepPartial } from 'redux';
+import { AdminConfig, ServiceSettings } from '@mattermost/types/config';
 
-export default class SessionLengthSettings extends AdminSettings {
-    getConfigFromState = (config) => {
+interface SessionLengthSettingsState extends BaseState {
+    extendSessionLengthWithActivity: ServiceSettings['ExtendSessionLengthWithActivity'];
+    sessionLengthWebInHours: ServiceSettings['SessionLengthWebInHours'];
+    sessionLengthMobileInHours: ServiceSettings['SessionLengthMobileInHours'];
+    sessionLengthSSOInHours: ServiceSettings['SessionLengthSSOInHours'];
+    sessionCacheInMinutes: ServiceSettings['SessionCacheInMinutes'];
+    sessionIdleTimeoutInMinutes: ServiceSettings['SessionIdleTimeoutInMinutes'];
+    sessionIdleTimeoutMobileInMinutes: number;
+}
+
+interface SessionLengthSettingsProps extends BaseProps {
+    license: {
+        Compliance: string;
+    };
+}
+
+export default class SessionLengthSettings extends AdminSettings<SessionLengthSettingsProps, SessionLengthSettingsState> {
+    getConfigFromState = (config: DeepPartial<AdminConfig>) => {
         const MINIMUM_IDLE_TIMEOUT = 5;
 
-        config.ServiceSettings.ExtendSessionLengthWithActivity = this.state.extendSessionLengthWithActivity;
-        config.ServiceSettings.SessionLengthWebInHours = this.parseIntNonZero(this.state.sessionLengthWebInHours);
-        config.ServiceSettings.SessionLengthMobileInHours = this.parseIntNonZero(this.state.sessionLengthMobileInHours);
-        config.ServiceSettings.SessionLengthSSOInHours = this.parseIntNonZero(this.state.sessionLengthSSOInHours);
-        config.ServiceSettings.SessionCacheInMinutes = this.parseIntNonZero(this.state.sessionCacheInMinutes);
-        config.ServiceSettings.SessionIdleTimeoutInMinutes = this.parseIntZeroOrMin(this.state.sessionIdleTimeoutInMinutes, MINIMUM_IDLE_TIMEOUT);
+        config.ServiceSettings!.ExtendSessionLengthWithActivity = this.state.extendSessionLengthWithActivity;
```

Comment:
Using non-null assertion seems not correct to me. Because on line 34 we are saying that some of the config can be null/ undefined by declaring the type with `DeepPartial<AdminConfig>`. Then we are saying it is non-null. 

We either change `DeepPartial<AdminConfig>` to `AdminConfig` OR better handle null situations with null checks